src/components/tl/ucp/allgather/allgather_knomial.c

```diff
@@ -230,14 +238,22 @@ ucc_status_t ucc_tl_ucp_allgather_knomial_init_r(
     ucc_base_coll_args_t *coll_args, ucc_base_team_t *team,
     ucc_coll_task_t **task_h, ucc_kn_radix_t radix)
 {
+    ucc_tl_ucp_team_t *tl_team = ucc_derived_of(team, ucc_tl_ucp_team_t);
     ucc_tl_ucp_task_t *task;
+    ucc_sbgp_t        *sbgp;
 
     task = ucc_tl_ucp_init_task(coll_args, team);
+    if (tl_team->cfg.use_reordering &&
+        coll_args->args.coll_type == UCC_COLL_TYPE_ALLREDUCE) {
```

Comment:
Can you explain why we only enable it for ALLREDUCE?

Reply:
allgather_knomial is used also for small message allgather and large message bcast (bcast SAG). In these cases reorder ranks will not be used yet - additional changes will need to be made.
Currently in this PR, the only usage of reorder ranks in allgather that is implemented is as part of allreduce SRA.